Fix Issue #5 and plug some memory leaks

MadeOfMagicAndWires opened the pull request:
Fixes #5 by allocating enough space for the full path (up to BUFFER_SIZE + 24) and also frees some leftover allocated memory.

Ideally, I'd like for the PAGES_BASE_DIR etc. to be configurable, but that is for another day.

MadeOfMagicAndWires committed:
Fix Issue #5 and plug some memory leaks

MadeOfMagicAndWires commented on shortcut.c:
```diff
@@ -275,7 +288,7 @@ int getFileContent(char const *path, char **out)
         goto error;
 
-    *out = (char *)malloc(len);
+    *out = (char *)calloc(len+1, sizeof(char));
```
Using calloc because valgrind complains when calling strlen on uninitialized memory.

mt-empty commented:
Looks good, thank you for your contribution.

> Ideally, I'd like for the PAGES_BASE_DIR etc. to be configurable, but that is for another day.

Same here.

MadeOfMagicAndWires commented:
Fixed smaller than logic which I'm sure I had written down correctly at *some* point. 

I think that was the last of it though.

MadeOfMagicAndWires commented:
Fixed another bug (in my brain); strncat != strncpy. 

That's really the last of it though, I'm going to get some breakfast.

(I also rebased the branch to keep the log clean)
